Sources/PostgresNIO/New/PostgresChannelHandler.swift

```diff
@@ -664,6 +688,92 @@ final class PostgresChannelHandler: ChannelDuplexHandler {
         }
     }
 
+    private func makePrepareStatementAction(
+        preparedStatement: PreparedStatementContext,
+        context: ChannelHandlerContext
+    ) -> PSQLTask {
+        let promise = self.eventLoop.makePromise(of: RowDescription?.self)
+        promise.futureResult.whenComplete { result in
+            switch result {
+            case .success(let rowDescription):
+                self.prepareStatementComplete(
+                    name: preparedStatement.name,
+                    rowDescription: rowDescription,
+                    context: context
+                )
+            case .failure(let error):
+                let psqlError: PSQLError
+                if let error = error as? PSQLError {
+                    psqlError = error
+                } else {
+                    psqlError = .connectionError(underlying: error)
```

Comment:
@fabianfett we said to wrap the `Error` in a `PSQLError`.

All the cases I see here
```swift
        enum Base: Sendable, Hashable {
            case sslUnsupported
            case failedToAddSSLHandler
            case receivedUnencryptedDataAfterSSLRequest
            case server
            case messageDecodingFailure
            case unexpectedBackendMessage
            case unsupportedAuthMechanism
            case authMechanismRequiresPassword
            case saslError
            case invalidCommandTag

            case queryCancelled
            case tooManyParameters
            case clientClosesConnection
            case clientClosedConnection
            case serverClosedConnection
            case connectionError
            case uncleanShutdown

            case listenFailed
            case unlistenFailed
        }
```
are quite specific. Is there any case I should use to wrap a generic error? Inspecting error to gather more information could be tricky